cmd/nerdctl/container_cp_linux.go

```diff
@@ -67,208 +54,28 @@ Using 'nerdctl cp' with untrusted or malicious containers is unsupported and may
 }
 
 func cpAction(cmd *cobra.Command, args []string) error {
-	srcSpec, err := parseCpFileSpec(args[0])
-	if err != nil {
-		return err
-	}
-
-	destSpec, err := parseCpFileSpec(args[1])
-	if err != nil {
-		return err
-	}
+	options, err := processCpOptions(cmd)
 
-	flagL, err := cmd.Flags().GetBool("follow-link")
 	if err != nil {
 		return err
 	}
 
-	if (srcSpec.Container != nil && destSpec.Container != nil) || (len(srcSpec.Path) == 0 && len(destSpec.Path) == 0) {
-		return fmt.Errorf("one of src or dest must be a local file specification")
-	}
-	if srcSpec.Container == nil && destSpec.Container == nil {
-		return fmt.Errorf("one of src or dest must be a container file specification")
-	}
-	if srcSpec.Path == "-" {
-		return fmt.Errorf("support for reading a tar archive from stdin is not implemented yet")
-	}
-	if destSpec.Path == "-" {
-		return fmt.Errorf("support for writing a tar archive to stdout is not implemented yet")
-	}
-
-	container2host := srcSpec.Container != nil
-	var container string
-	if container2host {
-		container = *srcSpec.Container
-	} else {
-		container = *destSpec.Container
-	}
 	ctx := cmd.Context()
 
-	// cp works in the host namespace (for inspecting file permissions), so we can't directly use the Go client.
-	selfExe, inspectArgs := globalFlags(cmd)
-	inspectArgs = append(inspectArgs, "container", "inspect", "--mode=native", "--format={{json .Process}}", container)
-	inspectCmd := exec.CommandContext(ctx, selfExe, inspectArgs...)
-	inspectCmd.Stderr = os.Stderr
-	inspectOut, err := inspectCmd.Output()
-	if err != nil {
-		return fmt.Errorf("failed to execute %v: %w", inspectCmd.Args, err)
-	}
-	var proc native.Process
-	if err := json.Unmarshal(inspectOut, &proc); err != nil {
-		return err
-	}
-	if proc.Status.Status != containerd.Running {
-		return fmt.Errorf("expected container status %v, got %v", containerd.Running, proc.Status.Status)
-	}
-	if proc.Pid <= 0 {
-		return fmt.Errorf("got non-positive PID %v", proc.Pid)
-	}
```

Comment:
I would prefer keeping this logic within `cmd` side:

1. They rely on calling `inspectCmd`, which means if we move it to `pkg` side, this `pkg` code implicitly depends on cli.
2. If it's moved to `pkg` side, we should not assume it's used in cli.

Reply:
Probably we should save `container2host, pid, dst, src, followSymlink` in `ContainerCpOptions` instead of `NerdctlCmd, NerdctlArgs`.

Reply:
Make sense, I'll do it.
I wasn't clear about the exact boundaries of xxxOptions before(Does it just contain the flags extracted **directly** from flags directly or may include  some **indirectly** generated vars for the real runner.)

Seems like there is no such limitation on xxxOptions, your suggestion is great.

Reply:
@djdongjin 
This way the final logic will remain very similar to the origin ── just an additional Options and the separated Kopy func, which is redundant? 
Maybe I should just close the PR and check the cp refactoring box?

Reply:
Anyway,
I followed the suggested refactor way.
Hope this helps. 🙏🏻